### PR TITLE
Add alternate mario-kart style(?) controller binding

### PR DESCRIPTION
**Pull request by cspotcode**
Only differences in the alternate layout: A/B and triggers are swapped.  Triggers do shooting, A/B do gas and brake.

I bound it to a `constants.py` flag.  I suppose it could also be bound to one of the new `debug_*` inputs from #36 if we wanted to toggle during gameplay.

### CODE REVIEW

**Comment by cspotcode** [comment]
[Add an alternative control scheme](https://trello.com/c/VK0ON0Qe/87-add-an-alternative-control-scheme)

File: src/constants.py [hunk 1]
```diff
@@ -14,3 +14,10 @@
 Also fixes known issue where, when pausing in debugger, update and draw each get
 called twice back-to-back instead of alternating once per.
 """
+
+USE_ALTERNATE_CONTROLLER_LAYOUT = False
+"""
+Use alternate (Mario Kart-style?) controller layout where A/B are gas and brake,
+and triggers are primary/secondary fire.  At this early stage of development,
+this flag is meant for experimentation.
+"""
```
Comment (Thorakii) [inline src/constants.py:23]: I personally like the idea of binding this to a debug button toggle for now while we're testing things out, but that's just an opinion. Non blocking

**Review by Thorakii — CHANGES_REQUESTED** [review]
Throws this error and won't run when `USE_ALTERNATE_CONTROLLER_LAYOUT = True`:
'XInputController' object has no attribute 'buttons'

I believe the issue has to do with the triggers being `VirtualAxis` and a/b being `VirturalButtons`

File: src/constants.py (hunk 1 above)
Comment (cspotcode) [reply src/constants.py:23]: How about dpad down?  I can merge the debug buttons PR first, then fixup this one to use dpad down.

I like keeping the boolean here so I can set it and never need to remember to hit the dpad debug button when I start and re-start the game.

**Comment by cspotcode** [comment]
Addressing feedback also uncovered some bugs in the input code which I've fixed.  Full list of changes:

**Notable changes**
Pressing `debug_2` (dpad down, 2 on keyboard) toggles between input schemes at runtime
Show control scheme's name to debug overlay
Rename constants flag to indicate it starts with that layout, since now can be toggled at runtime

**Less exciting stuff**
Fix bug in using controller buttons to drive virtual axes
Fix bug where toggling between control schemes meant triggers were doing *both* driving and shooting
Fixed crash when using controller w/VirtualAxis unbound to a real axis
Switch from `!= None` to `is not None` because the latter is idiomatic (perhaps faster? not sure)

File: src/constants.py (hunk 1 above)
Comment (Thorakii) [reply src/constants.py:23]: Awesome, having both options sounds great

**Review by Thorakii — APPROVED** [review]
Everything looks good and works now

**Pull request merged**